src/tcgwars/logic/impl/gen3/LegendMaker.groovy

```diff
@@ -1604,9 +1620,9 @@ public enum LegendMaker implements LogicCardInfo {
           text "Whenever you attach a React Energy card from your hand to Tangela, remove all damage counters from Tangela."
           delayedA {
             after ATTACH_ENERGY, self, {
-              if (ef.reason==PLAY_FROM_HAND && ef.card.name.contains("React Energy")) {
+              if (ef.reason==PLAY_FROM_HAND && ef.card.name == "React Energy") {
```

Comment:
I don't think this will work right. You may need to check that the card is in hand in a `before PLAY_ENERGY` to ensure it works properly in all cases. (See [Crazy Code Porygon-Z](https://tcgone.net/cards/porygon-z-unb-157) ability which should still activate this effect when used)

Reply:
Crazy code seems to attach as a PLAY_FROM_HAND, doesn't it? Tested it with Combusken from Dragon Majesty and it removed SPC no problem. Some cards do need improved checks tho, as in not asking for basic energies but that's for another commit/PR.

Reply:
Hmm, my understanding was that wasn't working right on Blastoise ex, but it looks like that wasn't using `attachEnergy` but was instead using `attachEnergyFrom`.